src/openms/include/OpenMS/FORMAT/FASTAFile.h

```diff
@@ -45,180 +45,168 @@
-    std::ofstream outfile_; ///< filestream for writing; init using FastaFile::writeStart()
-    std::unique_ptr<void, std::function<void(void*) > > reader_; ///< filestream for reading; init using FastaFile::readStart(); needs to be a pointer, since its not copy-constructable; we use void* here, to avoid pulling in seqan includes
-    Size entries_read_; ///< some internal book-keeping during reading
-  };
+    class OPENMS_DLLAPI FASTAFile
+    {
+    public:
+        /**
+            @brief FASTA entry type (identifier, description and sequence)
+            The first String corresponds to the identifier that is
+            written after the > in the FASTA file. The part after the
+            first whitespace is stored in description and the text
+            from the next line until the next > (exclusive) is stored
+            in sequence.
+          */
+        struct FASTAEntry
+        {
+            String identifier;
+            String description;
+            String sequence;
+
+            FASTAEntry() :
+                    identifier(),
+                    description(),
+                    sequence()
+            {
+            }
+
+            FASTAEntry(String id, String desc, String seq) :
+                    identifier(id),
+                    description(desc),
+                    sequence(seq)
+            {
+            }
+
+            FASTAEntry(const FASTAEntry& rhs)
+                    :
+                    identifier(rhs.identifier),
+                    description(rhs.description),
+                    sequence(rhs.sequence)
+            {
+            }
+
+            FASTAEntry(FASTAEntry&& rhs) noexcept
+                    :
+                    identifier(::std::move(rhs.identifier)),
+                    description(::std::move(rhs.description)),
+                    sequence(::std::move(rhs.sequence))
+            {
+            }
+
+            FASTAEntry& operator=(const FASTAEntry& rhs)
+            {
+                if (*this == rhs) return *this;
+                identifier = rhs.identifier;
+                description = rhs.description;
+                sequence = rhs.sequence;
+                return *this;
+            }
+
+            bool operator==(const FASTAEntry& rhs) const
+            {
+                return identifier == rhs.identifier
+                       && description == rhs.description
+                       && sequence == rhs.sequence;
+            }
+
+            bool headerMatches(const FASTAEntry& rhs) const
+            {
+                return identifier == rhs.identifier &&
+                       description == rhs.description;
+            }
+
+            bool sequenceMatches(const FASTAEntry& rhs) const
+            {
+                return sequence == rhs.sequence;
+            }
+        };
+
+        /// Default constructor
+        FASTAFile();
+
+        /// Destructor
+        virtual ~FASTAFile();
+
+        bool readSequence(std::string & seq);
```

Comment:
this function should be in the private section and thus be named ` ... readSequence_(...`